### PR TITLE
Remove keyword table

### PR DESCRIPTION
**Pull request by pitag-ha**
That's a lighter API. And anyways, all users of `Lexer.keyword_table` only use the keyword table to check if a string is an OCaml keyword.

To be merged after #226.

### CODE REVIEW

**Comment by pitag-ha** [comment]
There are currently two users of `Lexer.keyword_table`. I will send a patch to them after this PR gets approved.

File: src/keyword.ml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      "lor";
+      "lxor";
+      "mod";
+      "land";
+      "lsl";
+      "lsr";
+      "asr";
+    ]
+  in
+  List.mem ~set:keywords word
```
Comment (jeremiedimino) [inline src/keyword.ml:65]: Given that we only use the list inside this function, we could write as follow:

```
let is_keyword = function
  | "and" | "as" | ... | "asr" -> true
  | _ -> false
```

which would make it much faster.

**Review by jeremiedimino — CHANGES_REQUESTED** [review]
Looks good

File: src/keyword.ml (hunk 1 above)
Comment (pitag-ha) [reply src/keyword.ml]: Right, I didn't think about that. Thanks!

**Comment by pitag-ha** [comment]
I've sent patches to the two libraries that use `Lexer.keyword_table` at the moment: [one](https://github.com/janestreet/ppx_accessor/pull/3) to ppx_accessor and [one](https://github.com/janestreet/ppx_variants_conv/pull/9) to ppx_variants_conv

**Pull request merged**